commercial/app/controllers/PrebidAnalyticsController.scala

```diff
@@ -31,33 +31,23 @@ class PrebidAnalyticsController(val controllerComponents: ControllerComponents)
       .build()
   }
 
-  private val newLine = "\n".getBytes
+  private val charset = "UTF-8"
 
-  private def streamAnalytics(json: JsValue) = {
-    val record  = new Record().withData(ByteBuffer.wrap(toBytes(json) ++ newLine))
+  private def streamAnalytics(analytics: String) = {
+    val record  = new Record().withData(ByteBuffer.wrap(s"$analytics\n".getBytes(charset)))
     val request = new PutRecordRequest().withDeliveryStreamName(prebidAnalyticsStream).withRecord(record)
     val result  = firehose.putRecordFuture(request)
     result.failed foreach {
-      case NonFatal(e) => log.error(s"Failed to put '$json'", e)
+      case NonFatal(e) => log.error(s"Failed to put '$analytics'", e)
     }
   }
 
-  private def serve404[A](implicit request: Request[A]) =
-    Cached(CacheTime.NotFound)(WithoutRevalidationResult(NotFound))
-
-  def insert(): Action[JsValue] = Action(parse.json) { implicit request =>
+  def insert(): Action[String] = Action(parse.text) { implicit request =>
     if (prebidAnalytics.isSwitchedOn) {
       if (isProd) streamAnalytics(request.body)
-      else log.info(prettyPrint(request.body))
+      else log.info(prettyPrint(Json.parse(request.body)))
```

Comment:
Sometimes, computing the argument to log.<whatever> can be intensive, which is useless if the log level is not turned on. Is there a way to do something akin to:
```
log.info( blabla ) if log.isInfo
```
?

Reply:
This is true in Java, but in Scala the argument is lazy if you look at the signature of the `info` method in https://github.com/playframework/playframework/blob/master/framework/src/play/src/main/scala/play/api/Logger.scala#L151-L158
So no need to do the check in advance.